src/fsharp/fsi/fsi.fs

```diff
@@ -1120,23 +1119,23 @@ type internal FsiDynamicCompiler
 
         let filteredByVal =
             allNames
-            |> List.choose (function
```

Comment:
THis code is still kind of strange (as it was in the originakl version). It reports the top-level names defined, e.g. "x" in "let x = 1", but doesn't report any modules or types defined.

That's not related to this PR but the feature should eventually be adjusted to do a more systematic job on reporting freshly defined items.